File: app/src/main/java/com/example/petscoffee/database/CoffeeShopDao.java
package com.example.petscoffee.database;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.example.petscoffee.coffeeShop.CoffeeShop;

import java.util.ArrayList;
import java.util.List;

@Dao
public interface CoffeeShopDao {
    @Insert
    void insertCoffee(CoffeeShop coffee);

    @Query("SELECT * FROM coffeeShop WHERE id =:id")
    CoffeeShop queryCoffee(int id);

    @Query("SELECT * FROM coffeeShop WHERE account=:account")
    CoffeeShop queryCoffee(String account);

    @Delete
    void deleteCoffee(CoffeeShop coffee);

    @Query("SELECT * FROM coffeeShop")
    List<CoffeeShop> queryCoffees();

    @Update
    void upDateCoffee(CoffeeShop coffee);

    //上面为CoffeeShop数据的curd
}

File: app/src/main/java/com/example/petscoffee/repository/local/CoffeeShopDao.kt
package com.example.petscoffee.repository.local

import com.example.petscoffee.model.CoffeeShop
import androidx.lifecycle.LiveData
import androidx.room.*
import com.example.petscoffee.model.pets.Pets

@Dao
interface CoffeeShopDao {
    @Insert
    fun insertCoffee(coffee: CoffeeShop)

    @Query("SELECT * FROM coffeeShop WHERE id =:id")
    fun queryCoffee(id: Int): CoffeeShop

    @Query("SELECT * FROM coffeeShop WHERE account=:account")
    fun queryCoffee(account: String): CoffeeShop

    @Query("SELECT * FROM coffeeShop WHERE id=:id")
    fun queryCoffeeLiveData(id: Int): LiveData<CoffeeShop>

    @Delete
    fun deleteCoffee(coffee: CoffeeShop)

    @Query("SELECT * FROM coffeeShop")
    fun queryCoffees(): MutableList<CoffeeShop>

    @Update
    fun upDateCoffee(coffee: CoffeeShop) //上面为CoffeeShop数据的curd
}